### PR TITLE
Add support for querying sourcemax statistics

### PR DESCRIPTION
**Pull request by infinity0**
This allows us to more easily get some reasonable (although not entirely accurate) data about source packages.

### CODE REVIEW

File: src/popcon.py [hunk 1]
```diff
@@ -89,19 +89,14 @@
 
 # week in seconds
 EXPIRY = 86400 * 7
-DUMPFILE = os.path.join(
```
Comment (venthur) [inline src/popcon.py:92]: Why did you move that part to `_package_raw_generic`?

**Comment by venthur** [comment]
Hi @infinity0 thanks for the patch! Could you elaborate a bit more on what you did, so I can review the patch better? Thank you!

**Comment by infinity0** [comment]
Hi! I added some functions to get stats for source packages, rather than binary packages. These are read from the "sourcemax" statistics that popcon publishes in a different file - which is why I made the dumpfile a variable, since we now download and cache two files. (It is not possible to calculate this directly from all-popcon-results - it contains no information about source packages.)

The "sourcemax" statistic is the maximum score out of all the scores for the binary packages of that source package. This is a rough underapproximation of the more accurate "number of people that have installed binary-package-0 OR binary-package-1 OR .. OR binary-package-n", but unfortunately popcon doesn't give this latter number. But the popcon data is already inaccurate for lots of other reasons, so I thought it was OK to simply call this the "source_package score", and we could make it more accurate in the future if that becomes important.

**Comment by infinity0** [comment]
Hi Bastian, any news on this?

**Comment by infinity0** [comment]
ping

**Comment by venthur** [comment]
@infinity0 @lamby Thank you for the patch, and sorry for the long delay in my response!
